### DISCUSSION TITLE
[v6.0.1] Error: [requireEntitiesArray] Explicit list of entities is required, please use the 'entities' option.

### DISCUSSION THREAD

**Discussion by tsvetann** [mikro-orm/mikro-orm#5100] [Q&A] [answered]
```javascript
[info] MikroORM version: 6.0.1
[discovery] ORM entity discovery started, using ReflectMetadataProvider
[discovery] - processing 0 files
[discovery] - entity discovery finished, found 0 entities, took 8 ms
[discovery] ORM entity discovery started, using ReflectMetadataProvider in sync mode
Error: [requireEntitiesArray] Explicit list of entities is required, please use the 'entities' option.
.....
```

my config:

```javascript
await MikroORM.init({
        entities: ['path/with/no/entities'],
        entitiesTs: ['path/with/no/entities'],
        baseDir: process.cwd(),
        migrations: {
          snapshot: false,
          path:"path/with/no/migrations",
          emit: 'ts',
        },
        discovery: {
          warnWhenNoEntities: false,
          requireEntitiesArray: false,
          disableDynamicFileAccess: false,
        },
});
```

Any idea what I misconfigured so that I keep getting the error: `Error: [requireEntitiesArray] Explicit list of entities is required, please use the 'entities' option`. I know that there are no entities in the path provided as expected but I thought that `warnWhenNoEntities: false,` would mute the error, but instead I got requireEntitiesArray, which I am unable to get rid of with `disableDynamicFileAccess: false`. Thx

**Comment by B4nan** [comment, marked as answer]
> ORM entity discovery started, using ReflectMetadataProvider in sync mode

You are using synchronous discovery, which always requires entities array - folder-based discovery is not supported in this mode.

> await MikroORM.init({

You are clearly using `initSync` instead of `init`, otherwise this wouldn't happen. If it's not the case, I need to see a complete reproduction.
